apps/sms/test/marionette/generators/thread.js

```diff
@@ -5,48 +5,116 @@
 // TODO(gaye): This is no bueno and we need to fix.
 var _ = require('SMS/node_modules/lodash');
 
+var DEFAULT_THREAD_PARAMETERS = {
+  numberOfMessages: 1,
+  messageType: 'sms',
+  participants: ['+123'],
+  body: 'Thread message content'
+};
+
+var DEFAULT_MESSAGE = {
+  iccId: null,
+  sender: null,
+  delivery: 'sent',
+  timestamp: 0,
+  sentTimestamp: 0,
+  read: false
+};
+
+var DEFAULT_SMS = _.assign({
+  type: 'sms',
+  body: 'Thread message content',
+  receiver: null,
+  messageClass: 'class-1',
+  deliveryStatus: 'not-applicable',
+  deliveryTimestamp: 0
+}, DEFAULT_MESSAGE);
+
+var DEFAULT_MMS = _.assign({
+  type: 'mms',
+  receivers: [],
+  deliveryInfo: [{ deliveryStatus: 'not-applicable' }],
+  subject: 'Message subject',
+  smil: null,
+  attachments: [],
+  expiryDate: 0,
+  readReportRequested: false
+}, DEFAULT_MESSAGE);
+
 var ThreadGenerator = {
   uniqueThreadId: 0,
   uniqueMessageId: 0,
 
   generate: function(parameters) {
-
-    var DEFAULT_PARAMETERS = {
-      numberOfMessages: 1,
-      messageType: 'sms',
-      participants: ['+123'],
-      body: 'Thread message content'
-    };
-
-    parameters = _.assign(DEFAULT_PARAMETERS, parameters);
+    parameters = _.assign({}, DEFAULT_THREAD_PARAMETERS, parameters);
 
     var thread = {
       id: ++ThreadGenerator.uniqueThreadId,
       body: parameters.body,
       lastMessageType: parameters.messageType,
-      timestamp: Date.now(),
+      timestamp: 0,
       messages: [],
       participants: parameters.participants
     };
 
     for (var i = 0; i < parameters.numberOfMessages; i++) {
-      var id = ++ThreadGenerator.uniqueMessageId;
-      thread.messages.push({
-        id: id,
-        iccId: null,
-        threadId: thread.id,
-        sender: null,
-        receiver: parameters.participants[0],
-        type: parameters.messageType,
-        delivery: 'sent',
-        body: thread.body + ' (messageId: ' + id + ')',
-        timestamp: Date.now()
-      });
+      var message = null;
+      var messageBody = thread.body + ' (messageId: ' +
+        (ThreadGenerator.uniqueMessageId + 1) + ')';
+      if (parameters.messageType === 'sms') {
+        message = this.generateSMS({
+          threadId: thread.id,
```

Comment:
It seems here is too much repetition, but I didn't want to mix real message properties with ThreadGenerator parameters and it's for tests only after all :)